pkg/util/clusterdata/ingress_profiles_task.go

```diff
@@ -0,0 +1,129 @@
+	//             External -> Public
+	ingressControllers, err := ef.opClient.OperatorV1().IngressControllers("openshift-ingress-operator").List(ctx, metav1.ListOptions{})
+	if err != nil {
+		ef.log.Error(err)
+		errs <- err
+		return
+	}
+
+	//List Services from openshift-ingress namespace
+	//Among those Services, look for the ones of type LoadBalancer, with name "router-xxx". The matching will be done with
+	//IngressController name based on this xxx.
+	//Service IP will be taken from the candidate and added to the corresponding IngressProfile
+	services, err := ef.kubClient.CoreV1().Services("openshift-ingress").List(ctx, metav1.ListOptions{})
+	if err != nil {
+		ef.log.Error(err)
+		errs <- err
+		return
+	}
+
+	routerIPs := make(map[string]string)
+	for _, service := range services.Items {
+		svcName := service.ObjectMeta.Name
+		if len(svcName) < 8 || !strings.HasPrefix(svcName, "router-") || len(service.Status.LoadBalancer.Ingress) == 0 {
+			continue
+		}
+		matchingICName := svcName[7:] //router-xxx
+		routerIPs[matchingICName] = service.Status.LoadBalancer.Ingress[0].IP
+	}
+
+	//Reconcile IngressController and corresponding router service
+	ingressProfiles := make([]api.IngressProfile, len(ingressControllers.Items))
+	for i, ingressController := range ingressControllers.Items {
+		ingressProfiles[i] = api.IngressProfile{
+			Name: ingressController.ObjectMeta.Name,
+		}
+
+		// set IP
+		if IP, ok := routerIPs[ingressProfiles[i].Name]; ok {
+			ingressProfiles[i].IP = IP
+		} else {
+			//Unexpected, can't associate an IP from router service
+			ef.log.Infof("Failed to retrieve an IP from Router service for IngressController %q", ingressProfiles[i].Name)
+		}
+
+		// set Visibility
+		var visibility api.Visibility
+		if ingressController.Spec.EndpointPublishingStrategy == nil {
+			//Default case on Azure, LoadBalancerStrategy with External scope
+			//See https://docs.openshift.com/container-platform/4.6/networking/ingress-operator.html#nw-ingress-controller-configuration-parameters_configuring-ingress
+			visibility = api.VisibilityPublic
+		} else {
+			if ingressController.Spec.EndpointPublishingStrategy.LoadBalancer == nil {
+				//Non LoadBalancer strategy case
+				ef.log.Infof("Failed to get Visibility for IngressController %q", ingressProfiles[i].Name)
+			} else {
+				switch ingressController.Spec.EndpointPublishingStrategy.LoadBalancer.Scope {
+				case operatorv1.InternalLoadBalancer:
+					visibility = api.VisibilityPrivate
+				case operatorv1.ExternalLoadBalancer:
+					visibility = api.VisibilityPublic
+				default:
+					//Unexpected at this point
+					ef.log.Infof("Failed to get Visibility for IngressController %q", ingressProfiles[i].Name)
+				}
+			}
+		}
+
+		ingressProfiles[i].Visibility = visibility
+	}
+
+	callbacks <- func() {
+		ef.oc.Properties.IngressProfiles = ingressProfiles
+	}
+}
+
+func (ef *ingressProfileEnricherTask) SetDefaults() {
+	ef.oc.Properties.IngressProfiles = nil
```

Comment:
This means that if enrich fails we will not show anything to the user.
Should we do this or show "database model"? @jim-minter?

Reply:
I'm +1 on this, as it is